pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -622,24 +640,33 @@ private void searchForCandidateBroker(NamespaceBundle bundle,
     }
 
     private boolean isBrokerActive(String candidateBroker) {
-        List<String> brokers = pulsar.getLocalMetadataStore().getChildren(LoadManager.LOADBALANCE_BROKERS_ROOT).join();
-
-        for (String brokerHostPort : brokers) {
-            if (candidateBroker.equals("http://" + brokerHostPort)) {
+        URI uri = URI.create(candidateBroker);
+        String candidateBrokerHostAndPort = uri.getHost() + ":" + uri.getPort();
+        Set<String> availableBrokers = getAvailableBrokers();
+        for (String brokerHostPort : availableBrokers) {
+            if (candidateBrokerHostAndPort.equals(brokerHostPort)) {
                 if (LOG.isDebugEnabled()) {
-                    LOG.debug("Broker {} found for SLA Monitoring Namespace", brokerHostPort);
+                    LOG.debug("Broker {} is available.", brokerHostPort);
                 }
                 return true;
             }
         }
 
         if (LOG.isDebugEnabled()) {
-            LOG.debug("Broker not found for SLA Monitoring Namespace {}",
-                    candidateBroker + ":" + config.getWebServicePort());
+            LOG.debug("Broker {} couldn't be found in available brokers {}",
+                    candidateBroker, availableBrokers);
         }
         return false;
     }
 
+    private Set<String> getAvailableBrokers() {
+        try {
+            return loadManager.get().getAvailableBrokers();
+        } catch (Exception e) {
```

Comment:
can we avoid catching "Exception" ?

Reply:
It should prob get converted into a `CompletableFuture<List<String>> getAvailableBrokersAsync()`

Reply:
@merlimat I guess that shouldn't be done as part of this PR? I'd like to have this PR on branch-2.8 and branch-2.9 too to improve the topic ownership lookups.